pkg/operation/botanist/controlplane.go

```diff
@@ -1158,17 +1126,68 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
 			foundStatefulset = false
 		}
 
+		if foundEtcd && etcd.Status.Etcd.Name != "" {
```

Comment:
Move this before `Get` call for statefulset. And may be to not conofuse anymore with use `name` variable. Change it to `stsName`.